components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListener.java

```diff
@@ -165,9 +176,40 @@ public boolean doPreSetUserClaimValuesWithID(String userID, Map<String, String>
         Map<String, String> scimToLocalMappings = SCIMCommonUtils.getSCIMtoLocalMappings();
         String modifiedLocalClaimUri = scimToLocalMappings.get(SCIMConstants.CommonSchemaConstants.LAST_MODIFIED_URI);
         claims.put(modifiedLocalClaimUri, lastModifiedDate);
+
+        // Validate dob value against the regex.
+        validateDateOfBirthClaimValue(claims, userStoreManager);
```

Comment:
Let's validate this for doPreSetUserClaimValueWithID as well.

Reply:
fixed in 3d8d84d